### ISSUE TITLE
pymqi maked exe to connect IBM MQ failed

### ISSUE THREAD

**Issue by Smallfi** [dsuch/pymqi#115]
hello，IBM engineers,
I am a user in China, I meet a problem too difficult to reslove
when I use Python 3.7 and  pymqi lib 1.9.2
to  connect IBM MQ 9.1

I run Python code with Eclipse ,  Push the message to the MQ successfully.
Then I use python Pyinstaller3.4  lib ,use command (Pyinstaller -F -w testQM.py) to make an exe on windows platform.
I run the exe programme with console , meet the problem, in the picture below
![image](https://user-images.githubusercontent.com/7116142/58530990-53987280-8213-11e9-9104-4079891b1603.png)

**Comment by JoshMcMQ** [comment]
Did you execute the exe on the same computer?

**Comment by Smallfi** [comment]
I execute the exe on the same computer

**Comment by Smallfi** [comment]
is there anyone here can help me about this thing?

**Comment by chughts** [comment]
Eclipse will have set up an environment for you - paths and lib paths. The error you are getting suggests that you don't have the same environment set up when you run the .exe

**Comment by SeyfSV** [comment]
Can you trace your application with strmqtrc?

**Comment by Smallfi** [comment]
> Eclipse will have set up an environment for you - paths and lib paths. The error you are getting suggests that you don't have the same environment set up when you run the .exe

but can you tell me what to do to reslove the problem

**Comment by Smallfi** [comment]
> Can you trace your application with strmqtrc?

I do not know what you said,  are you good at using pymqi?

**Comment by SeyfSV** [comment]
Read [this](https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_8.0.0/com.ibm.mq.ref.adm.doc/q083660_.htm)
Run this: strmqtrc -t all -p <exe_file_name>
Attach file with your program pid from `mq_installation_path/trace folder`
Also you can search 2012 in this file and try to found cause of the issue.

PS
May be I am not good in using pymqi or creating exe files from .py files, but I good enough in MQ troubleshooting.

**Comment by Smallfi** [comment]
> Read [this](https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_8.0.0/com.ibm.mq.ref.adm.doc/q083660_.htm)
> Run this: `strmqtrc -t all -p <exe_file_name>`
> Attach file with your program pid from `mq_installation_path/trace folder`
> Also you can search 2012 in this file and try to found cause of the issue.
> 
> PS
> May be I am not good in using pymqi or creating exe files from .py files, but I good enough in MQ troubleshooting.
![image](https://user-images.githubusercontent.com/7116142/59016178-43b40a80-8873-11e9-92d8-5fdc4afeee7e.png)
tell me i have not the priviledge to  use this command

**Comment by Smallfi** [comment]
> Read [this](https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_8.0.0/com.ibm.mq.ref.adm.doc/q083660_.htm)
> Run this: `strmqtrc -t all -p <exe_file_name>`
> Attach file with your program pid from `mq_installation_path/trace folder`
> Also you can search 2012 in this file and try to found cause of the issue.
> 
> PS
> May be I am not good in using pymqi or creating exe files from .py files, but I good enough in MQ troubleshooting.

I  think when i make the exe , the runtime Environment has changed,which is different from the Eclipse Environment

**Comment by SeyfSV** [comment]
> tell me i have not the priviledge to use this command

Run cmd with privileges: right mouse button -> Run as Administrator..., and then run MQ commands.

Also you cat try to setup environment before running  your .exe :  [setmqenv](https://www.ibm.com/support/knowledgecenter/en/SSFKSJ_8.0.0/com.ibm.mq.ref.adm.doc/q083560_.htm)

**Comment by KYDronePilot** [comment]
I had pretty much the exact same issue. I am using MQ client version `9.0.4.0-IBM-MQC-Win64` on Windows 10-1903.

The actual issue appeared to be caused by MQ treating `C:\Users\<active_user>\IBM\MQ` as if it were `C:\Program Files\IBM\MQ`. Putting in a symlink fixed that particular issue, but then other issues came up.

We finally determined that if we excluded the `mqe.dll` and `mqic.dll` files from being automatically bundled by PyInstaller, it fixed both this issue and the other issues we were getting.

This can be accomplished in PyInstaller by including the following in a custom `.spec` file just after the `a = Analysis(...)` section (more info on `.spec` files in PyInstaller can be found [here](https://pythonhosted.org/PyInstaller/spec-files.html)):

```python
a.binaries = a.binaries - TOC(
    [
        ('mqe.dll', None, None),
        ('mqic.dll', None, None)
    ]
)
```
